crates/biome_js_analyze/src/utils/restricted_glob.rs

```diff
@@ -7,27 +7,46 @@ use biome_rowan::{TextRange, TextSize};
 /// - Use `\*` to escape `*`
 /// - `?`, `[`, `]`, `{`, and `}` must be escaped using `\`.
 ///   These characters are reserved for future use.
-/// - `!` must be escaped if it is the first character of the pattern
+/// - Use `!` as first character to negate the glob
 ///
 /// A path segment is delimited by path separator `/` or the start/end of the path.
 #[derive(Clone, Debug, serde::Deserialize, serde::Serialize)]
 #[serde(try_from = "String", into = "String")]
-pub struct RestrictedGlob(globset::GlobMatcher);
+pub struct RestrictedGlob {
+    is_negated: bool,
+    glob: globset::GlobMatcher,
+}
 impl RestrictedGlob {
+    /// Returns `true` if this glob is negated.
+    pub fn is_negated(&self) -> bool {
+        self.is_negated
+    }
+
     /// Tests whether the given path matches this pattern or not.
     pub fn is_match(&self, path: impl AsRef<std::path::Path>) -> bool {
-        self.0.is_match(path)
+        self.is_raw_match(path) != self.is_negated
+    }
+
+    /// Tests whether the given path matches this pattern or not ignoring the negation.
```

Comment:
```suggestion
    /// Tests whether the given path matches this pattern, ignoring the negation
```

When you use the word "whether", the "or not" is always implied by the word itself